src/arviz_plots/plots/utils.py

```diff
@@ -295,3 +295,124 @@ def add_reference_lines(
+):
+    """Add reference bands.
+
+    This function adds lines to a plot collection based on the provided
+    references. It supports both vertical and horizontal lines, depending on the
+    specified orientation.
+
+    Parameters
+    ----------
+    plot_collection : PlotCollection
+        Plot collection to which the reference lines will be added.
+    references : tuple, list or dict
+        Reference values to be plotted as bands/shaded regions.
+    orientation : str, default "vertical"
+        The orientation of the reference lines, either "vertical" or "horizontal".
+    aes_map : mapping of {str : sequence of str}, optional
+        Mapping of artists to aesthetics that should use their mapping in `plot_collection`
+        when plotted. Valid keys are the same as for `plot_kwargs`.
+
+        The default is to use an "overlay" aesthetic for all elements.
+
+        It is possible to request aesthetics without mappings defined in the
+        provided `plot_collection`. In those cases, a mapping of "ref_dim" to the requested
+        aesthetic will be automatically added.
+    plot_kwargs : mapping of {str : mapping or False}, optional
+        Valid keys are:
+
+        * "ref_band" -> passed to :func:`~arviz_plots.visuals.vspan` for vertical `orientation`
+          and to :func:`~arviz_plots.visuals.hspan` for horizontal `orientation`
+
+    sample_dims : list, optional
+        Dimensions to reduce unless mapped to an aesthetic.
+        Defaults to ``rcParams["data.sample_dims"]``
+    ref_dim : list, optional
+        Specifies the name of the reference dimension for reference values.
+    **kwargs : mapping of {str : sequence}, optional
+        Mapping of aesthetic keys to the values to be used in their mapping.
+        See :func:`~arviz_plots.PlotCollection.generate_aes_dt` for more details.
+
+    Returns
+    -------
+    plot_collection : PlotCollection
+        Plot collection with the reference lines added.
+
+    Examples
+    --------
+    Add reference bands from 0 to 2 and 3 to 5 for all variables.
+
+    .. plot::
+        :context: close-figs
+
+        >>> from arviz_plots import plot_dist, add_reference_bands, style
+        >>> style.use("arviz-variat")
+        >>> from arviz_base import load_arviz_data
+        >>> dt = load_arviz_data('centered_eight')
+        >>> pc = plot_dist(
+        >>>     dt,
+        >>>     kind="ecdf"
+        >>> )
+        >>> add_reference_bands(pc, references=[[0, 2], [3, 5]])
+    """
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map["ref_band"] = aes_map.get("ref_band", ["overlay_ref"])
+    if sample_dims is None:
+        sample_dims = list(set(plot_collection.data.dims).difference(plot_collection.facet_dims))
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    if ref_dim is None:
+        ref_dim = ["ref_dim_0", "ref_dim_1"]
```

Comment:
As this will later on be passed as input to `references_to_dataset` and some of them will also show in the artist/aesthetic mappings generated, I think it would be better to use more meaningful names (even if they were longer). Here for example, if we consider each band to be a "single reference" these would be `ref_dim, band_dim`.